lib/test.js

```diff
@@ -154,8 +154,18 @@ Test.prototype.run = function () {
 		}
 
 		ret.then(
-			function () {
-				self.exit();
+			function (assertRet) {
+				var exitReason;
+
+				if (assertRet instanceof Error) {
+					exitReason = assertRet;
+				}
+
+				if (assertRet && assertRet.length) {
```

Comment:
I think `if (assertRet.length)` would do just fine on it's own, no need for `assertRet &&`.